crates/sqllib/src/string.rs

```diff
@@ -8,47 +14,62 @@ use crate::{
 
 use core::fmt::Error;
 use feldera_types::{deserialize_without_context, serialize_without_context};
-#[cfg(not(feature = "arcstring"))]
+#[cfg(feature = "interned_string")]
 use internment::ArcIntern;
 use like::{Escape, Like};
 use regex::Regex;
-use rkyv::Fallible;
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::{Deserialize, Serialize};
 use size_of::{Context, SizeOf};
 use std::{
-    borrow::Cow,
     cmp::{max, min},
     fmt::{Display, Formatter},
     sync::Arc,
 };
 
-#[cfg(not(feature = "arcstring"))]
+#[cfg(feature = "interned_string")]
 type StringRef = ArcIntern<String>;
 
-#[cfg(feature = "arcstring")]
+#[cfg(not(feature = "interned_string"))]
 type StringRef = Arc<String>;
 
-#[derive(Clone, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
+/// An immutable reference counted string.
+#[derive(
+    Clone,
+    Default,
+    Debug,
+    Eq,
+    Hash,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    rkyv::Archive,
```

Comment:
do these work for interned?
should they be feature-gated?

Reply:
They don't I think. This probably won't compile with interned strings.